interface/src/app/gene/genes.js

```diff
@@ -0,0 +1,384 @@
+  return {
+    controller: ['$scope', 'SelectedGenesFactory', '$state',
+      function($scope, SelectedGenesFactory, $state) {
+        $scope.selectedGenes = SelectedGenesFactory.return_genes();
+
+        $scope.sendToNetwork = function() {
+          var genes = [];
+          for (var gene in $scope.selectedGenes) {
+            genes.push(gene);
+          }
+          $state.go('gene_network', {
+            'genes': genes
+          });
+        };
+      }
+    ],
+    replace: true,
+    restrict: "E",
+    scope: true,
+    templateUrl: 'gene/selected-genes-panel.tpl.html'
+  };
+})
+
+// Directive for button where user does not find a result
+// Should remove entire row from list
+.directive('noResultButton', ['SearchResults', function(SearchResults) {
+  return {
+    link: function(scope, element, attr) {
+      element.bind("click", function() {
+        // Because this results in data being updated
+        // and isn't using ng-* listeners we need to
+        // wrap things in $apply()
+        // http://jimhoskins.com/2012/12/17/angularjs-and-apply.html
+        scope.$apply(function() {
+          SearchResults.remove(scope.query);
+        });
+      });
+    },
+    replace: true,
+    restrict: "E",
+    scope: {
+      query: '@'
+    },
+    templateUrl: 'gene/no-result-button.tpl.html'
+  };
+}])
+
+// Directive for button with a gene, should add gene
+// and remove entire row from list
+.directive('geneResultButton', ['SearchResults', 'SelectedGenesFactory',
+  function(SearchResults, SelectedGenesFactory) {
+    return {
+      restrict: "E",
+      link: function(scope, element, attr) {
+        element.bind("click", function() {
+          scope.$apply(function() {
+            SelectedGenesFactory.addGene(scope.gene);
+            SearchResults.remove(scope.query);
+          });
+        });
+      },
+      replace: true,
+      templateUrl: 'gene/gene-result-button.tpl.html'
+    };
+  }
+])
+
+// Directive for button with a gene, should add gene
+// and remove entire row from list
+.directive('selectedGeneButton', ['SelectedGenesFactory',
+  function(SelectedGenesFactory) {
+    return {
+      restrict: "E",
+      link: function(scope, element, attr) {
+        element.bind("click", function() {
+          scope.$apply(function() {
+            SelectedGenesFactory.removeGene(scope.gene);
+          });
+        });
+      },
+      replace: true,
+      templateUrl: 'gene/gene-result-button.tpl.html'
+    };
+  }
+])
+
+// Directive for button to get more options, should get
+// next page of search results for this query from the server
+.directive('moreResultButton', ['SearchResults', function(SearchResults) {
+  return {
+    link: function(scope, element, attr) {
+      element.bind("click", function() {
+        scope.page = scope.pageDict.page + 1;
+        scope.$apply(function() {
+          scope.pageDict.page = scope.page;
+          scope.updatePage(scope.page);
```

Comment:
What does `updatePage` do for us that hasn't already been handled with updates to `scope.page` and `scope.pageDict.page`? This doesn't feel Angular-ish.